crypto/sm2/sm2_crypt.c

```diff
@@ -200,17 +216,19 @@ int ossl_sm2_encrypt(const EC_KEY *key,
         goto done;
     }
 
-    msg_mask = OPENSSL_zalloc(msg_len);
-    if (msg_mask == NULL)
-       goto done;
-
     /* X9.63 with no salt happens to match the KDF used in SM2 */
     if (!ossl_ecdh_kdf_X9_63(msg_mask, msg_len, x2y2, 2 * field_size, NULL, 0,
                              digest, libctx, propq)) {
         ERR_raise(ERR_LIB_SM2, ERR_R_EVP_LIB);
         goto done;
     }
 
+    if (is_all_zeros(msg_mask, msg_len)) {
+        memset(msg_mask, 0, msg_len);
```

Comment:
I don't think this memset is necessecary.  If is_all_zeros returns 1, then you are guaranteed that msg_mask is already zero

Reply:
Yes, you're  right. I'll fix this.